File: src/ImplementQueryByTwoStacks/Queue.java
public class Queue {
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public Queue() {
       // do initialization if necessary
       stack1 = new Stack<>();
       stack2 = new Stack<>();
    }
    
    public void push(int element) {
        // write your code here
        stack1.push(element);
    }

    public int pop() {
        // write your code here
        if(stack2.isEmpty()){
            putStack1ToStack2();
        }
        return stack2.pop();
    }

    public int top() {
        // write your code here
        if(stack2.isEmpty()){
            putStack1ToStack2();
        }
        return stack2.peek();
    }
    
    private void putStack1ToStack2(){
        while(!stack1.isEmpty()){
                stack2.push(stack1.pop());
        }
    }
}

File: src/ImplementQueryByTwoStacks/Queue.cpp
class Queue {
public:
    stack<int> stack1;
    stack<int> stack2;

    Queue() {
        // do intialization if necessary
    }

    void push(int element) {
        // write your code here
        stack1.push(element);
    }
    
    int pop() {
        // write your code here
        adjust();
        int tmp = stack2.top();
        stack2.pop();
        return tmp;
    }

    int top() {
        // write your code here
        adjust();
        return stack2.top();
    }
    
    void adjust(){
        if(stack2.empty()){
            while(!stack1.empty()){
                stack2.push(stack1.top());
                stack1.pop();
            }
        }
    }
};
